person.rb

```diff
@@ -1,22 +1,38 @@
+require './nameable'
+require './rental'
 # the person class
-class Person
+class Person < Nameable
   attr_reader :id
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
 
-  def initialize(age, name: 'Unkown', parent_permission: true)
+  def initialize(age, name = 'Unkown', parent_permission: true)
+    super
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
+  def self.all
+    ObjectSpace.each_object(self).to_a
+  end
+
   private
```

Comment:
While you did a good job working on the project requirements. Your work seems to give an error whenever I try to create a person. 

Take a look at the screenshot below:
![Screenshot 2023-11-10 at 11 40 33 AM](https://github.com/patriciachrysy/oop-school-library/assets/79506105/1b8d6d2e-cacd-480e-995f-8307e0f48f42)

Please fix this issue and make sure you test the console app to see if everything works before you submit for review.

Reply:
Though i noticed the error, the screenshots do not seem to match my codes, please make sure you test my code next time.Thank you> I made some fixes and changes thanks for reviewing again